dev/janus-discord.js

```diff
@@ -0,0 +1,164 @@
+const ws = new WebSocket(`wss://gateway.discord.gg/?v=8&encoding=json`)
+
+const ONCE = {}
+const ON = {}
+let s = null
+let intents = 0
+let heartbeatAc
+const heartbeat = () => {
+  console.log('heartbeat')
+  heartbeatAc = setTimeout(() => {
+    console.log('the client was ZOMBIFED')
+    console.log('TODO: try too reconnect+resume!')
+    Deno.exit(1)
+  }, 1000)
+  ws.send(JSON.stringify({ op: 1, d: s }))
+}
+
+ws.addEventListener('close', (event) => {
+  console.log(event)
+  Deno.exit(1)
+})
+
+ws.addEventListener('message', (event) => {
+  const { t, s, op, d } = JSON.parse(event.data)
```

Comment:
discord send messages with
- `t` (what **type**)
- `op` (what kind of **operation**)
- `s` (some message identifier, used for controlling integrity of the bot)
- `d` (JSON data)